### PR TITLE
CLDIDE-2636: Improve account model

### PR DESCRIPTION
**Pull request by evoevodin**
In this PR i: 
* Improved account model
* Added `AccountDao.exists` method
* Fixed `WorkspaceServiceClient` behaviour, it was sending 'null' string instead of null object in query parameters.

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ok

**Comment by akorneta** [comment]
ok

File: platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/server/AccountService.java [hunk 1]
```diff
@@ -158,9 +158,7 @@ public Response create(@Context SecurityContext securityContext,
         } catch (NotFoundException ignored) {
         }
         final String accountId = NameGenerator.generate(Account.class.getSimpleName().toLowerCase(), Constants.ID_LENGTH);
-        final Account account = new Account().withId(accountId)
-                                             .withName(newAccount.getName())
-                                             .withAttributes(newAccount.getAttributes());
+        final Account account = new Account(accountId, newAccount.getName());
```
Comment (garagatyi) [inline platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/server/AccountService.java]: why you omit setting account attributes?

File: platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/server/dao/AccountDao.java [hunk 2]
```diff
@@ -151,4 +154,50 @@ default boolean isWorkspaceRegistered(String workspaceId) throws ServerException
...
+     *         when {@code accountId} is null
+     * @throws ServerException
+     *         when any other error occurs
+     * @see #getById(String)
+     */
+    default boolean exist(@NotNull String accountId) throws ServerException {
+        try {
+            getById(requireNonNull(accountId, "Required non-null account id"));
+            return true;
+        } catch (NotFoundException ignored) {
```
Comment (garagatyi) [inline platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/server/dao/AccountDao.java]: It is not ignored exception

File: platform-api/che-core-api-account/src/test/java/org/eclipse/che/api/account/AccountServiceTest.java [hunk 3]
```diff
@@ -368,7 +352,7 @@ public void shouldBeAbleToGetAccountByName() throws Exception {
     }
 
     @DataProvider(name = "roleProvider")
-    public String[][] roleProvider() {
+    public Object[][] roleProvider() {
```
Comment (garagatyi) [inline platform-api/che-core-api-account/src/test/java/org/eclipse/che/api/account/AccountServiceTest.java]: Can you explain why you changed that?

File: platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/server/AccountService.java (hunk 1 above)
Comment (evoevodin) [inline platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/server/AccountService.java]: Its missrefactoring, thank you

File: platform-api/che-core-api-account/src/test/java/org/eclipse/che/api/account/AccountServiceTest.java (hunk 3 above)
Comment (evoevodin) [inline platform-api/che-core-api-account/src/test/java/org/eclipse/che/api/account/AccountServiceTest.java]: `@DataProvider` javadoc says:  _The annotated method must return an `Object[][]` where each `Object[]` can be assigned the parameter list of the test method._

**Comment by garagatyi** [comment]
ok

**Pull request merged**